### DISCUSSION TITLE
HSR initialization issue: possibly hardware issue?

### DISCUSSION THREAD

**Discussion by Yoo-Youngjae** [HumanSupportRobot/QandA#11] [Q&A]
We made our HSR can boot thanks to your help, but unfortunately we are in another trouble again.
Our HSR head display gets stuck on initial screen when booted.
We are sure that network is not a cause this time, and we doubt that this is a hardware problem.

![KakaoTalk_Image_2022-07-14-23-36-10](https://user-images.githubusercontent.com/44061214/179033038-cf487fe8-c74c-4a1a-b891-21d10226c8c2.jpeg)

We executed bash of docker containers, and we got above screenshot in `docker.hsrb.robot.service` container.
It seems that `hsrb_robot_service` script executes `exxx_read_hash` binary file, but it did not proceeds further.
Rather, `exxx_read_hash /dev/ttyCTI2 11` process repeatedly died and executed.
We doubt that a hardware related `ttyCTI2` is broken or unplugged, but we are not sure.
Is it really related with any hardware issues? Or could be there any possible reasons?

**Comment by Yoo-Youngjae** [comment]
we entered the following command.
`exxx_hash /dev/ttyCTI2 11`
But there is no response, only `Connection timed out`.

**Reply by hsr-admin** [reply]
Log into the robot using administrator privileges, then set it to not automatically start up using the following command:

`$ sudo systemctl stop docker.hsrb.robot.service`

Release the stop button, then, after waiting about 10 seconds

Please execute the following "exxx_hash" command:
`$ exxx_hash /dev/ttyCTI2 11`

**Reply by Yoo-Youngjae** [reply]
No response, same as above.

**Reply by hsr-admin** [reply]
Possible causes of malfunction include amplifier failure, blown fuse, or poor wiring contact.
The only things that can be done locally are to check for blown fuses and disconnected connectors.
If the amplifier is faulty, it will have to be replaced, so it will be difficult to restore the system immediately.

The following is a good reference for amplifier failure analysis
https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_motor_driver.html#id1

**Reply by Yoo-Youngjae** [reply]
After robocup, I checked again, and it seems that the motorboard problem is correct. 
Could you tell me how we can purchase the same motorboard?

Thank you.

**Reply by hsr-admin** [reply]
### Possible Cause

The cause could be a blown fuse, amplifier failure, or faulty wiring.
First, please perform the following two steps and let us know the results.

### Blown fuse check

　Check for a blown fuse by referring to the following manual:
　https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_motor_driver.html#id1

### Confirmation of amplifier failure
　Depending on the location of the amplifier failure, communication with all amplifiers may become impossible.
　First, check communication with all amplifiers using the following command:

`　$ exxx_hash /dev/ttyCTI2 <joint_id>`

　If communication with all amplifiers is not possible, remove the amplifiers downstream in the daisy chain in order and
　Run the following command to identify the amplifiers that cannot communicate.

`　$ exxx_hash /dev/ttyCTI2 <joint_id>`

**Reply by Yoo-Youngjae** [reply]
1. Fuse is not blown.
2. When we remove the 1110 cable for checking 'joint_id 41', we can see the response for all board except 41 axis board.

So we think, the board in hand(41 axis board) is in problem.

And then, can we have some solution for repairing it?

below image is our hand board.
![KakaoTalk_20220805_163255495](https://user-images.githubusercontent.com/44061214/183026183-5d0f1c26-53a9-463a-a3dc-a2bf6cc59be4.jpg)

**Reply by hsr-admin** [reply]
The motor amplifier needs to be replaced.
The contact information for the repair person in your group is listed below.

    hsr-support@robocup.or.jp

Contact them to receive a replacement motor amplifier.

This motor amplifier is not commercially available and is not generally available, so it is provided free of charge.

**Reply by hsr-admin** [reply]
Please paste here the output of the following command on any axis to check the firmware version.
`$ exxx_hash /dev/ttyCTI2 <joint_id>`

**Reply by Yoo-Youngjae** [reply]
![IMG_6731](https://user-images.githubusercontent.com/44061214/183339924-dabae4bb-eb0d-4059-93dc-8b6bc58eb6da.JPG)

here is the output of the "exxx_hash /dev/ttyCTI2 11"

**Reply by Yoo-Youngjae** [reply]
We received the motor amplifier for hand(41 axis board) and replace it.
And we also did clean install using iso file as below.
"https://packages.hsr.io/images/hsrb-installer_21.10.01_focal_HSRC-PHASE1_en_US.UTF-8.iso"

Now, hsr wake up after releasing emergency-stop.
It's very good news for us.
But the commands like rostopic, ihsrb, exxx_hash not work.

However, strangely, rgb image and point cloud is running normally at http://hsrb.local/user/, and even all motor operations are running normally at http://hsrb.local/user/.

According to the diagnostic at "http://hsrb.local/admin/", there is error in the "computer/file".

The same problem occurred even though I did the clean install twice.
Please tell me how to control the robot with ros, ihsrb, etc.

Thanks.

**Reply by hsr-admin** [reply]
> "https://packages.hsr.io/images/hsrb-installer_21.10.01_focal_HSRC-PHASE1_en_US.UTF-8.iso"

This file is the ISO image file for HSRC.

Your robot is HSRB, so please use the following ISO file:
` hsrb-installer_22.04.01_focal_HSRB-PHASE5_en_US.UTF-8.iso`

**Reply by Yoo-Youngjae** [reply]
Sorry. We used "hsrb-installer_22.04.01_focal_HSRB-PHASE5_en_US.UTF-8.iso". not hsrb-installer_21.10.01_focal_HSRC-PHASE1_en_US.UTF-8.iso.

But same issue occurred.

**Reply by hsr-admin** [reply]
The exxx_hash command requires the package to be installed on the local host of HSR as follows
```
$ sudo apt-get update
$ sudo apt-get install ros-noetic-hsrb-full
```

 "computer/file". Other HSRs sometimes generate errors, but they do not seem to affect the operation, so please use them as they are.


Note that ihsrb can be used from a PC by setting ROS_MASTER_URI.
hsrb_mode automatically sets the ROS_MASTER_URI.

**Reply by Yoo-Youngjae** [reply]
After downloading a ros-neotic-hsrb-full, it works successfully.

Thank you.

**Reply by LimSeunghyeon1** [reply]
Hello. We've done a diagnosis of our HSRB (with @Yoo-Youngjae ) via hsrb.local/admin

Here's  screenshots of it, it seems that our robot has problems with hand_camera, stereo_camera, head_center_camera, and imu sensors. 
We wonder if you can give any support or advices.
Thank you. 

![diagnosis1](https://user-images.githubusercontent.com/63534351/187688316-fe099429-b39a-431f-b654-5fd3549da163.png)
rs.
![diagnosis2](https://user-images.githubusercontent.com/63534351/187688328-5aca599e-f9a1-4ddd-9f0f-07b0591648f8.png)
![diagnosis3](https://user-images.githubusercontent.com/63534351/187688336-673fedcd-66c6-40e0-b4c2-f3e8050ce17b.png)

**Reply by hsr-admin** [reply]
Refer to the following documents to check the logs and see if the device is recognized, which may provide clues to hardware failure or other problems.

<https://docs.hsr.io/hsr_log_analysis_manual_en/log_files.html#id1>
<https://docs.hsr.io/hsr_log_analysis_manual_en/hsrb_others.html#usb>
<https://docs.hsr.io/hsr_faq_en/index.html>

**Reply by LimSeunghyeon1** [reply]
Hello. Thank you for your help. We've done diagnosis following above links. It seems that main devices mentioned in (https://docs.hsr.io/hsr_log_analysis_manual_en/hsrb_others.html#usb) are all connected properly. Here's error logs of journalctl and lsusb.
Unfortunately, we've got other big issues right now. Our robot had problem with its battery: it never get charged(even though we try to charge the battery), and shows ERROR sign on monitor instead of showing the percentage of the battery level. 


Thank you.

[hsr_error_log.zip](https://github.com/HumanSupportRobot/QandA/files/9555566/hsr_error_log.zip)

**Reply by hsr-admin** [reply]
The logs have been checked. Please investigate in the following order.

When 1. is resolved, check if it works properly, and if the battery problem still remains, please perform 2.

1. The device (Bus 003 Device 004: ID 0403:6001) is not recognized, please investigate according to the maintenance manual ([USB-IO Troubleshooting](https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_usb_io.html#usb-io)).

2. The battery is also not communicating with the CPU. Please investigate according to the maintenance manual ([Battery Failure Analysis Procedure](https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_battery.html#id1)).

**Reply by LimSeunghyeon1** [reply]
![battery](https://user-images.githubusercontent.com/63534351/191015682-8cbb4edd-d496-4f8f-8957-94f15ed0f241.png)
The battery seems working... although I somewhat feel that it charges slower than before....
I wonder why electric current indicates -3.24A in this picture(sanyo_battery_status -v command)? (In manual page it indicates 2.40A). Is it okay? I think It may be the reason why it charges slowly...

**Reply by LimSeunghyeon1** [reply]
Also, Bus 003 Device 004: ID 0403:6001 appears first, but suddenly disappears , leaving error messages RunStop I/O error on the monitor... We've changed wire 1002 to the new one, but the problem exists.

**Reply by hsr-admin** [reply]
The charging terminals may be degraded (carbonized), resulting in increased resistance, or the terminals may have been pushed back, resulting in poor contact.
Please visually check the charging terminals on the charger side and robot side, take pictures and upload them with reference to the following.

![charger](https://user-images.githubusercontent.com/103644635/191167336-a56d2470-98f8-4920-bec0-dbb52366efce.png)

**Reply by LimSeunghyeon1** [reply]
![KakaoTalk_20220923_185310901_04](https://user-images.githubusercontent.com/63534351/191936977-2d7cd7e8-15a9-461c-957c-e7f547bbe6bd.jpg)
![KakaoTalk_20220923_185310901_02](https://user-images.githubusercontent.com/63534351/191937185-0a62cadb-a657-44b9-bfb2-4c2bcb195eb8.jpg)
![KakaoTalk_20220923_185310901_01](https://user-images.githubusercontent.com/63534351/191937288-1abe30a5-dc78-4101-a60f-6abad5fefbf8.jpg)
![KakaoTalk_20220923_190444130__](https://user-images.githubusercontent.com/63534351/191938176-4f93b8c3-c1a7-4cdc-8f20-b9131ed953af.jpg)




Here's photos of the robot and the charger, it seems it doesn't get carbonized.

**Reply by hsr-admin** [reply]
Thanks for uploading the photos.
Judging from the photos, it looks like both the charger side and the robot side terminals are fine.

If you have more than one HSR, switch the battery, charger, and HSR itself to isolate the problem.
Please use electric_current to determine if the battery is charging properly.

(Reference)
　When charging normally, electric_current is about -5[A].
　https://docs.hsr.io/hsr_log_analysis_manual_en/hsrb_others.html#id2

**Reply by LimSeunghyeon1** [reply]
Thank you for the reply. We have only only single HSR robot, but we checked that the electric current is -5A.
We'll forget about the battery problem and focus on USB-IO issues.

Now our robot has problem of usb_io, force_torque, and front/back bumper according to the diagnosis.
We followed the manual pages
https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_usb_io.html
https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_force_sensor.html#id2

We've found out that usb_io, force_torch, and bumper are all related to USB-IO, and thought there will be a problem with communication between USB hub, USB converter card and CPU.

**We changed the wire 1002, 1016 by ourselves but nothing changed (based on the instructions from https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_usb_io.html)

Thus we're going to change USB converter and 1027 wire, so can we get these products from you?**

**Reply by hsr-admin** [reply]
For information on obtaining replacement parts that are not commercially available, please contact  
      hsr-support@robocup.or.jp

There is also a case in which recovery was achieved by replacing the USB hub in a similar case.
It may be a good idea to try replacing the USB hub since it is available on the market.

**Reply by LimSeunghyeon1** [reply]
Hello. Thank you for the reply all the times.
We switched the USB hub, but nothing changed, so we're  going to switch wire 1027 and USB converter as they delivered.
By the way, our model rarely wake up: usually they woke up normally right after we turn on the robot, but after we make e-stop and restart again it never woke up again.
So I wonder if this problem can be related to USB I/O issues.

**Reply by LimSeunghyeon1** [reply]
Hello. Hsr-admin.
We've done the diagnosis of USB I/O error based on (https://docs.hsr.io/hsrb_maintenance_manual_en/src/check_usb_io.html)

What we've done specifically are the following:
- Check 1002, 1016 wire on the body usb hub.
- Change the usb hub of the body and 1027 pin as well as the USB converter card that you sent to us.
- Check wide angle camera by 'cheese' -> **it doesn't work on the robot screen ( although it does show up on lsusb). However, when we check it on the other computer, it does work and we've concluded that the camera itself is okay.**
- Check the usb hub of the lifting axis -> we tried to get the right model and change it from the new one, but unfortunately the model (with the older version that is in our robot) is no longer available.
By the way, we think the usb hub of the body is working fine. **Because we notice it showed up on the other computer**(connecting with the wire 1015, which originally connects the hub of the lifting axis and the body). **We also confirm that 'cheese' works successfully.** 
- Check the wire 66, 64 is continuous and got 5v,12v by the multimeter.

We're really afraid that we keep asking you several times, but could you tell us other good solutions to solve this problem?

**Reply by hsr-admin** [reply]
Now, as a next step, please wait a moment while we prepare and send you a replacement power relay board.

**Reply by LimSeunghyeon1** [reply]
Thank you for the reply. Please notify us after you send the board.

**Reply by hsr-admin** [reply]
Due to export procedures, the power relay board will not be shipped until after December 6.
Please wait a little longer.

**Reply by LimSeunghyeon1** [reply]
Hello. Did the board ship without any problem?

**Reply by hsr-admin** [reply]
We apologize for the inconvenience.
Export procedures have been delayed and we have not been able to ship yet. Please wait a while.

**Reply by LimSeunghyeon1** [reply]
Hello, we've sent an email about shipping issue and our current progress to hsr_support team.
We apologize for our decision to return the power relay board.
We found out that the robot works well when switching to wireless mode but doesn't work on wired connection.

Thus, our next step is to make wired connection available.
We considered two options, dhcp and linked-only connection. (https://docs.hsr.io/hsrb_user_manual_en/howto/network_settings.html?highlight=avahi#hsr-network-check)

1) dhcp
we made a dhcp server in PC and wirely connect with a robot (as well as modifying /etc/network/interfaces.eth )
$ 
 
2) linked-only
we followed the steps in the manual, and make static ip by modifying /etc/network/interfaces.eth

**By 1,2, we are able to connect with hsrb.local by ssh however, the robot doesn't stand up at all.**

Also, we cannot1 see enp4s0::avahi in command $ ip address show as below.

$ ip address show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eno1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:30:64:4c:a2:19 brd ff:ff:ff:ff:ff:ff
    inet 10.255.255.1/24 brd 10.255.255.255 scope global eno1
       valid_lft forever preferred_lft forever
    inet6 fe80::230:64ff:fe4c:a219/64 scope link 
       valid_lft forever preferred_lft forever
**3: enp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:0c:8b:20:08:bc brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.8/24 brd 192.168.0.255 scope global dynamic enp4s0
       valid_lft 440sec preferred_lft 440sec
    inet6 fe80::20c:8bff:fe20:8bc/64 scope link 
       valid_lft forever preferred_lft forever**
4: wlp3s0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether f0:42:1c:c1:12:36 brd ff:ff:ff:ff:ff:ff
5: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:13:29:d6:b7 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever

---------------------------------------------------------------------------------------------

Can you offer us any advice on this problem?
Thank you.

**Comment by LimSeunghyeon1** [comment]
Dear hsr-admin,
We've solved the problems and now our robot works well at both wired and wireless mode.
The problem is on the internet service: a robot doesn't initialize when the internet cannot be configured.
Therefore, we solved this problem by install NAT in our PC.
We wonder if a robot actually need internet connection to initialize(perhaps it still remains other issues) ... but anyway it works well right now.
We deeply appreciate your help for several months.

Thank you.
